src/main/java/com/stripe/net/LiveStripeResponseGetter.java

```diff
@@ -109,6 +110,50 @@ public InputStream requestStream(
     return responseStream.body();
   }
 
+  @Override
+  public StripeResponse rawRequestStream(
+      ApiResource.RequestMethod method, String url, String content, RawRequestOptions options)
+      throws StripeException {
+    StripeRequest request = StripeRequest.createWithStringContent(method, url, content, options);
+
+    Map<String, String> additionalHeaders = options.getAdditionalHeaders();
+
+    if (additionalHeaders != null) {
+      for (Map.Entry<String, String> entry : additionalHeaders.entrySet()) {
+        String key = entry.getKey();
+        String value = entry.getValue();
+        request = request.withAdditionalHeader(key, value);
+      }
+    }
+
+    if (options.getApiMode() == ApiMode.PREVIEW) {
+      request = request.withAdditionalHeader("Stripe-Version", Stripe.PREVIEW_API_VERSION);
```

Comment:
Where is this value set for non-preview requests?

Reply:
It's set in `buildHeaders` when the `StripeRequest` is constructed: https://github.com/stripe/stripe-java/blob/beta/src/main/java/com/stripe/net/StripeRequest.java#L187

Reply:
Any reason not to handle this header in StripeRequest? We already have some apiMode-specific logic there.

Reply:
Good point, moved to StripeRequest